pinecone_rag/ingestion.py

```
@@ -391,96 +694,141 @@ def _upsert_batch(
+        processed, errors, failed_documents = 0, [], []
+        selected_keys = []
+        if process_type == ProcessType.UPDATE_DOCUMENT_BY_ID:
+            selected_keys = [k for k in records[0].keys() if k != "id"]
+        elif process_type == ProcessType.UPDATE_METADATA_BY_ID:
+            selected_keys = keys or [
+                k for k in records[0].keys() if k not in ("id", "chunk_text")
+            ]
+        elif process_type == ProcessType.UPDATE_VALUE_BY_ID:
+            selected_keys = ["chunk_text"]
+        else:
+            raise ValueError(f"Unsupported process type: {process_type}")
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)